### PR TITLE
Update documentation of action inputs/outputs

### PR DESCRIPTION
**Pull request by petecheslock**
Here's a first draft where i'm trying to basically replicate how [add-and-commit](https://github.com/EndBug/add-and-commit/blob/main/README.md) setup their README.md which is a nice standard for actions.

### CODE REVIEW

**Comment by petecheslock** [comment]
@kgilpin check this out and let me know if this is what you had in mind.

**Comment by kgilpin** [comment]
Yup this looks really nice! I think this will be a big step in clarity for people who want to understand beyond the setup tool. I’ll have a few comments soon.

File: README.md [hunk 1]
```diff
@@ -1,4 +1,6 @@
-# getappmap/install-appmap
+# getappmap/install-appmap <!-- omit in toc -->
```
Comment (kgilpin) [inline README.md:1]: This is used with auto TOC which is a nice VSCode feature

File: README.md [hunk 2]
```diff
@@ -1,4 +1,6 @@
...
+
+To get started with AppMap in GitHub actions, you need to start by installing the [AppMap App on the official GitHub Marketplace](https://github.com/marketplace/get-appmap)
```
Comment (kgilpin) [inline README.md:3]: It might look clean to put these two paragraphs:

> In a fence / offset like this

File: README.md [hunk 3]
```diff
@@ -9,38 +11,103 @@ This is a GitHub action to install and configure AppMap. It will do the followin
...
+    # Consult https://appmap.io/docs/add-appmap-to-a-project.html for more information.
+    project-type: 'pip'
+
+    # Command working directory
+    # Default: '.'
+    directory: /path/to/code
+    
+    # Contents of appmap.yml configuration.
+    # Default: Automatically generated appmap.yml file
+    appmap-config: custom-appmap.yml
```
Comment (kgilpin) [inline README.md:44]: This input would actually be the configuration text not the file name. So a multi line of YAML

File: README.md [hunk 4]
```diff
@@ -9,38 +11,103 @@ This is a GitHub action to install and configure AppMap. It will do the followin
...
+
+    # Build file to be configured, in case of ambiguity. This is an advanced option.
+    # Default: Automatically identified based on project language
+    build-file: requirements.txt
+    
+    # Installer name to be used, in case of ambiguity. This is an advanced option.
+    # Default: appmap
```
Comment (kgilpin) [inline README.md:51]: It’s so advanced we might want to leave it out. People can consult action.yml if they really want it.

File: README.md [hunk 5]
```diff
@@ -9,38 +11,103 @@ This is a GitHub action to install and configure AppMap. It will do the followin
...
+
+    # Add the .appmap directory to .gitignore, if it's not already present.
+    # Default: true
+    ignore-dot-appmap: false
+
+    # Install the AppMap command-line tools.
+    # Default: true
+    install-appmap-tools: false
+
+    # Install the and configure the AppMap language library. This can be set to false if your project already has AppMap libraries included in your project build dependency file.
```
Comment (kgilpin) [inline README.md:70]: GitHub UI isn’t adding line breaks automatically so we should have a max line width

File: README.md [hunk 6]
```diff
@@ -9,38 +11,103 @@ This is a GitHub action to install and configure AppMap. It will do the followin
...
+    # Default: true
+    build-patch-file: false
+
+    # Path specification to use when creating the patch file. If the patch file includes files that you don't want to commit, you can use this option to exclude them.
+    # Default: ". ':(exclude,top)vendor' ':(exclude,top)node_modules'"
+    diff-path-spec: "':(exclude,top)virtualenv'"
+
+    # Expected value of the appmap_dir in appmap.yml. If this input is provided, the action will verify that the configured appmap_dir matches the expected value. If the value does not match, the action will fail.
+    # Default: tmp/appmap
+    expected-appmap-dir: other/appmap/dir
```
Comment (kgilpin) [inline README.md:84]: I think we can omit this too since we now use the same appmap directory everywhere

File: README.md [hunk 7]
```diff
@@ -9,38 +11,103 @@ This is a GitHub action to install and configure AppMap. It will do the followin
...
+- name: Install AppMap tools
+  uses: getappmap/install-action@v1
+  with:
+    # Choose the type of your project here: e.g. bundler, maven, gradle, pip, pipenv, poetry, yarn, npm, etc.
+    # Consult https://appmap.io/docs/add-appmap-to-a-project.html for more information.
+    project-type: 'pip'
+
+    # Command working directory
+    # Default: '.'
+    directory: /path/to/code
```
Comment (kgilpin) [inline README.md:40]: Here we can explain that this option is for monorepo / multi project support. 

So the example should be a subproject path like `./projects/backend`

File: README.md [hunk 8]
```diff
@@ -9,38 +11,103 @@ This is a GitHub action to install and configure AppMap. It will do the followin
...
+    
+    # Contents of appmap.yml configuration.
+    # Default: Automatically generated appmap.yml file
+    appmap-config: custom-appmap.yml
+
+    # Build file to be configured, in case of ambiguity. This is an advanced option.
+    # Default: Automatically identified based on project language
```
Comment (kgilpin) [inline README.md:47]: Project language and build framework.

File: README.md (hunk 3 above)
Comment (petecheslock) [reply README.md:44]: Ahh - makes sense - [I tested this out here and looks good.](https://github.com/land-of-apps/rails_tutorial_sample_app_7th_ed/actions/runs/6591238305/workflow#L29-L34)

File: README.md (hunk 2 above)
Comment (petecheslock) [reply README.md:3]: Oh good idea!

File: README.md (hunk 5 above)
Comment (petecheslock) [reply README.md:70]: Ahh yea - good point, i'll play around to see what looks good

**Comment by petecheslock** [comment]
ok @kgilpin check this out - i've incorporated all your recommendations.   If you're good then go ahead and merge anytime.  I'll have a later PR to point here from https://appmap.io/docs/

**Review by kgilpin — APPROVED** [review]

**Pull request merged**